pkg/daemon/update.go

```diff
@@ -194,7 +196,19 @@ func (dn *Daemon) writeUnits(units []ignv2_2types.Unit) error {
 		if err != nil {
 			return fmt.Errorf("Failed to write systemd unit %q: %v", u.Name, err)
 		}
-
+		// if the unit should be enabled, then enable it.
+		// otherwise the unit is disabled. run disableUnit to ensure the unit is
+		// disabled. even if the unit wasn't previously enabled the result will
+		// be fine as disableUnit is idempotent.
+		if *u.Enabled {
```

Comment:
the semantics here for the `Enabled` field should be
1. is the pointer nil -> do nothing
2. is it true -> enable
3. is it false -> disable

we also have to deal with the legacy `Enable` field as well, which is just
1. is it true -> Enable
2. else do nothing